### PR TITLE
Fix array_intersect_key with non empty array

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes https://github.com/phpstan/phpstan/issues/10561

### CODE REVIEW

File: tests/PHPStan/Analyser/nsrt/array-intersect-key.php [hunk 1]
```diff
@@ -15,9 +15,9 @@ class Foo
 	public function nonEmpty(array $arr, array $arr2): void
 	{
 		assertType('non-empty-array<int|string, string>', array_intersect_key($arr));
-		assertType('non-empty-array<int|string, string>', array_intersect_key($arr, $arr));
+		assertType('array<int|string, string>', array_intersect_key($arr, $arr));
```
Comment (VincentLanglet) [inline tests/PHPStan/Analyser/nsrt/array-intersect-key.php:18]: I'm not sure it's worth adding complexity to the ArrayIntersectKeyFunctionReturnTypeExtension in order to check if all the variable are the same. This array_intersect_key  call is useless...

Comment (staabm) [reply tests/PHPStan/Analyser/nsrt/array-intersect-key.php:18]: The test is only about the types. No need for checking identical vars passed

Comment (VincentLanglet) [reply tests/PHPStan/Analyser/nsrt/array-intersect-key.php:18]: Then 
```
array_intersect_key($nonEmptyArray1, $nonEmptyArray2)
```
is now correctly understood as an array that may be empty.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**